### PR TITLE
rename callback FineTune arg `round`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

rename argument which is mimicking build in python function and making it potential danger

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/finetuning.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
 log = logging.getLogger(__name__)
 
 
-def multiplicative(epoch):
```
Comment (carmocca) [inline pytorch_lightning/callbacks/finetuning.py:35]: I'd keep the name so anybody reading the code knows what it is, can also add typing

File: pytorch_lightning/callbacks/finetuning.py [hunk 2]
```diff
@@ -354,8 +334,8 @@ def __init__(
         initial_denom_lr: float = 10.0,
         train_bn: bool = True,
         verbose: bool = False,
-        round: int = 12,
-    ):
+        rounding: int = 12,
```
Comment (carmocca) [inline pytorch_lightning/callbacks/finetuning.py:337]: No compatibility concerns about this rename?

Comment (Borda) [reply pytorch_lightning/callbacks/finetuning.py:337]: have checked and we do not user round as argument anywhere else...

File: pytorch_lightning/callbacks/finetuning.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/callbacks/finetuning.py]: reverted

File: pytorch_lightning/callbacks/finetuning.py [hunk 3]
```diff
@@ -393,10 +373,12 @@ def on_fit_start(self, trainer, pl_module):
             return super().on_fit_start(trainer, pl_module)
         raise MisconfigurationException("The LightningModule should have a nn.Module `backbone` attribute")
 
-    def freeze_before_training(self, pl_module: "pl.LightningModule"):
+    def freeze_before_training(self, pl_module: "pl.LightningModule") -> None:
         self.freeze(pl_module.backbone)
 
-    def finetune_function(self, pl_module: "pl.LightningModule", epoch: int, optimizer: Optimizer, opt_idx: int):
+    def finetune_function(
+        self, pl_module: "pl.LightningModule", epoch: int, optimizer: Optimizer, opt_idx: int
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/finetuning.py:380]:
```suggestion
        self, pl_module: "pl.LightningModule", epoch: int, optimizer: Optimizer, optimizer_idx: int
```
we have optimizer_idx in all other public methods in lightning.

File: pytorch_lightning/callbacks/finetuning.py [hunk 4]
```diff
@@ -366,12 +346,12 @@ def __init__(
...
         self.train_bn: bool = train_bn
         self.verbose: bool = verbose
-        self.round: int = round
+        self.rounding: int = rounding
         self.previous_backbone_lr: Optional[float] = None
 
     def on_save_checkpoint(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", checkpoint: Dict[str, Any]
-    ) -> Dict[int, Any]:
+    ) -> Dict[str, Any]:
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/finetuning.py:354]:
```suggestion
    ) -> dict:
```
doesn't necessarily be string key.

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/callbacks/finetuning.py [hunk 5]
```diff
@@ -294,45 +282,37 @@ def on_train_epoch_start(self, trainer, pl_module):
...
 
-    def freeze_before_training(self, pl_module: "pl.LightningModule"):
+    def freeze_before_training(self, pl_module: "pl.LightningModule") -> None:
         """Override to add your freeze logic."""
         raise NotImplementedError
 
 
 class BackboneFinetuning(BaseFinetuning):
-    r"""
+    r"""Finetune a backbone model based on a learning rate user-defined scheduling.
```
Comment (s-rog) [inline pytorch_lightning/callbacks/finetuning.py:297]:
```suggestion
    r"""Fine tune a backbone model based on a user-defined learning rate schedule.
```

File: pytorch_lightning/callbacks/finetuning.py [hunk 6]
```diff
@@ -294,45 +282,37 @@ def on_train_epoch_start(self, trainer, pl_module):
...
-
+            By default, we will use ``current_learning /  backbone_initial_ratio_lr``
         should_align: Wheter to align with current learning rate when backbone learning
             reaches it.
-
         initial_denom_lr: When unfreezing the backbone, the intial learning rate will
-            current_learning_rate /  initial_denom_lr.
-
+            ``current_learning_rate /  initial_denom_lr``.
         train_bn: Wheter to make Batch Normalization trainable.
```
Comment (s-rog) [inline pytorch_lightning/callbacks/finetuning.py:313]:
```suggestion
        train_bn: Whether to make Batch Normalization layers trainable.
```

**Review by s-rog — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
